pkg/pub_package_reader/lib/src/package_dir.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'tar_utils.dart';
+
+/// Helper methods to access package content.
+abstract class PackageDir {
```

Comment:
Can we call it `PackageFileSystem` or `PackageContents` or `PackageFolder` ?

Reply:
What exceptions can this throw, any? It's fine if it only throws `Error` then we don't have document it.

let's also document that the path format, we use slashes, probably without stating slash...